plasmapy/particles/particle_class.py

```diff
@@ -2494,7 +2492,7 @@ def molecule(
 # If ParticleLike is renamed or moves out of particle_class.py, check
 # for a link to its doc page in error messages in _factory.py.
 
-ParticleLike = Union[str, Integral, Particle, CustomParticle, u.Quantity]
+ParticleLike: TypeAlias = Union[str, Integral, Particle, CustomParticle, u.Quantity]  # noqa: UP007
```

Comment:
Needed to keep this as using `typing.Union` because using the `|` made it so that we couldn't redefine `ParticleLike.__doc__` below.